src/commands.ts

```diff
@@ -168,6 +169,8 @@ export async function registerPlugin(uri: vscode.Uri): Promise<void> {
 
 async function createDataPluginFromSampleFile(dataPluginName: string): Promise<DataPlugin | null> {
     const dataPlugin: DataPlugin = new DataPlugin(dataPluginName, 'hello_world', Languages.Python);
+    await dataPlugin.pluginIsInitialized();
```

Comment:
I think the root cause for this issue is that you call an async function in the constructor of `DataPlugin`. We could get rid of this by providing an async factory function to replace each constructor call. What do you think?

Reply:
Thanks Lukas for pushing me in the right direction 👍👍. Please check if my implementation proposal is what you expected.

Reply:
That's perfect!